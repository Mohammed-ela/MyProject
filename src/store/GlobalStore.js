// src/store/GlobalStore.js
import React, { createContext, useState } from 'react';

export const GlobalContext = createContext();

export const GlobalProvider = ({ children }) => {
  const [userInfo, setUserInfo] = useState({ name: '', surname: '' });

  return (
    <GlobalContext.Provider value={{ userInfo, setUserInfo }}>
      {children}
    </GlobalContext.Provider>
  );
};
