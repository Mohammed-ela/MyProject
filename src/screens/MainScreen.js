import React, { useState, useContext } from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet, Alert } from 'react-native';
import LottieView from 'lottie-react-native';
import { GlobalContext } from '../store/GlobalStore';

const MainScreen = ({ navigation }) => {
  const { setUserInfo } = useContext(GlobalContext);
  const [name, setName] = useState('');
  const [surname, setSurname] = useState('');
  const [isLoading, setIsLoading] = useState(false);

  const handleGenerateQRCode = () => {
    if (!name || !surname) {
      Alert.alert('Erreur', 'Veuillez remplir les champs nom et prénom.');
      return;
    }

    setIsLoading(true); // Active le chargement
    setTimeout(() => {
      setUserInfo({ name, surname }); // Enregistre les données dans le store
      setIsLoading(false); // Désactive le chargement
      navigation.navigate('QRCodeScreen'); // Navigue vers l’écran QR Code
    }, 2000); // Simule un délai pour la génération
  };

  return (
    <View style={styles.container}>
      {isLoading ? (
        <LottieView
          source={require('../assets/Lego.json')}
          autoPlay
          loop
          style={styles.lottie}
        />
      ) : (
        <>
          <Text style={styles.title}>Générer un QR Code</Text>
          <Text style={styles.label}>Nom :</Text>
          <TextInput
            style={styles.input}
            placeholder="Entrez votre nom"
            placeholderTextColor="#888"
            value={name}
            onChangeText={setName}
          />
          <Text style={styles.label}>Prénom :</Text>
          <TextInput
            style={styles.input}
            placeholder="Entrez votre prénom"
            placeholderTextColor="#888"
            value={surname}
            onChangeText={setSurname}
          />
          <TouchableOpacity style={styles.button} onPress={handleGenerateQRCode}>
            <Text style={styles.buttonText}>Générer le QR Code</Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
    padding: 16,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 20,
  },
  label: {
    fontSize: 16,
    fontWeight: '500',
    color: '#555',
    alignSelf: 'flex-start',
    marginLeft: '10%',
    marginBottom: 8,
  },
  input: {
    width: '80%',
    height: 40,
    borderColor: '#ccc',
    borderWidth: 1,
    borderRadius: 5,
    paddingHorizontal: 10,
    marginBottom: 16,
    backgroundColor: '#fff',
  },
  button: {
    backgroundColor: '#007bff',
    paddingVertical: 12,
    paddingHorizontal: 30,
    borderRadius: 25,
    alignItems: 'center',
    marginTop: 20,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '600',
  },
  lottie: {
    width: 200,
    height: 200,
  },
});

export default MainScreen;
