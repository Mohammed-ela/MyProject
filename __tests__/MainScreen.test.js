
import React from 'react';
import { act } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import MainScreen from '../src/screens/MainScreen';
import { GlobalProvider } from '../src/store/GlobalStore';
afterEach(() => {
  jest.clearAllMocks();
  jest.clearAllTimers();
});

jest.useFakeTimers();

describe('MainScreen Tests', () => {
  it('snapshot de l’écran principal', () => {
    const { toJSON } = render(
      <GlobalProvider>
        <MainScreen />
      </GlobalProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('le composant de l’écran principal existe', () => {
    const { getByText } = render(
      <GlobalProvider>
        <MainScreen />
      </GlobalProvider>
    );
    expect(getByText('Nom :')).toBeTruthy();
    expect(getByText('Prénom :')).toBeTruthy();
  });

  it('les champs de saisie de nom et prénom existent', () => {
    const { getByPlaceholderText } = render(
      <GlobalProvider>
        <MainScreen />
      </GlobalProvider>
    );
    expect(getByPlaceholderText('Entrez votre nom')).toBeTruthy();
    expect(getByPlaceholderText('Entrez votre prénom')).toBeTruthy();
  });

  it('le bouton permettant d’accéder à l’écran de QR Code existe', () => {
    const { getByText } = render(
      <GlobalProvider>
        <MainScreen />
      </GlobalProvider>
    );
    expect(getByText('Générer le QR Code')).toBeTruthy();
  });

  it('la fonction d’ouverture d’écran du QR Code est appelée une fois', () => {
    // Mock de la navigation
    const mockNavigation = { navigate: jest.fn() };
  
    // Rendu du composant
    const { getByText, getByPlaceholderText } = render(
      <GlobalProvider>
        <MainScreen navigation={mockNavigation} />
      </GlobalProvider>
    );
  
    // Simule la saisie des champs
    fireEvent.changeText(getByPlaceholderText('Entrez votre nom'), 'John');
    fireEvent.changeText(getByPlaceholderText('Entrez votre prénom'), 'Doe');
  
    // Simule l’appui sur le bouton
// Simule l’appui sur le bouton dans un bloc `act`
act(() => {
  fireEvent.press(getByText('Générer le QR Code'));
  jest.runAllTimers(); // Simule l’exécution des timers
});
    // Simule l'exécution des timers
    jest.runAllTimers();
  
    // Vérifie que la navigation est appelée une fois avec les bons paramètres
    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith('QRCodeScreen');
  });
  
});
