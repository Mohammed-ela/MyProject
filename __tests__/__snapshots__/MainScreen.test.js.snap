// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen Tests snapshot de l’écran principal 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Générer un QR Code
  </Text>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "fontWeight": "500",
        "marginBottom": 8,
        "marginLeft": "10%",
      }
    }
  >
    Nom :
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Entrez votre nom"
    placeholderTextColor="#888"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 16,
        "paddingHorizontal": 10,
        "width": "80%",
      }
    }
    value=""
  />
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "fontWeight": "500",
        "marginBottom": 8,
        "marginLeft": "10%",
      }
    }
  >
    Prénom :
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Entrez votre prénom"
    placeholderTextColor="#888"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 16,
        "paddingHorizontal": 10,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#007bff",
        "borderRadius": 25,
        "marginTop": 20,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Générer le QR Code
    </Text>
  </View>
</View>
`;
